src/Host/Broker/Impl/RemoteUri/RemoteUriHelper.cs

```diff
@@ -1,78 +1,154 @@
         public static async Task HandlerAsync(HttpContext context) {
-            BinaryReader reader = new BinaryReader(context.Request.Body);
-            var url = reader.ReadString();
-            var method = reader.ReadString();
-            Dictionary<string, string> headers = JsonConvert.DeserializeObject<Dictionary<string, string>>(reader.ReadString());
+            var url = context.Request.Headers["x-rtvs-url"];
+
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+            request.Method = context.Request.Method;
+            request.UseDefaultCredentials = true;
+            SetRequestHeaders(request, context.Request.Headers);
+
+            if (context.Request.ContentLength > 0) {
+                Stream reqStream = await request.GetRequestStreamAsync();
+                await context.Request.Body.CopyToAsync(reqStream);
+                await reqStream.FlushAsync();
+            }
 
-            HttpClient client = new HttpClient();
-            Uri uri = new Uri(url);
+            HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync();
 
-            HttpMethod httpMethod = new HttpMethod(method);
-            HttpRequestMessage req = new HttpRequestMessage(httpMethod, uri);
+            SetResponseHeaders(response, context.Response);
 
-            foreach (var key in headers.Keys) {
-                var val = headers[key];
-                req.Headers.Add(key, val);
+            Stream respStream = response.GetResponseStream();
+            await respStream.CopyToAsync(context.Response.Body);
+            await context.Response.Body.FlushAsync();
+        }
+
+        private static long GetLong(string value) {
+            long ret;
+            if (long.TryParse(value, out ret)) {
+                return ret;
             }
+            return 0;
+        }
 
-            Stream content = null;
-            if (httpMethod == HttpMethod.Post) {
-                content = new MemoryStream();
-                long length = context.Request.Body.Length - context.Request.Body.Position;
-                var buffer = reader.ReadBytes((int)length);
-                await content.WriteAsync(buffer, 0, buffer.Length);
-                await content.FlushAsync();
-                content.Position = 0;
+        private static DateTime GetDateTime(string value) {
+            DateTime ret;
+            if (DateTime.TryParse(value, out ret)) {
+                return ret;
+            }
+            return new DateTime();
+        }
 
-                req.Content = new StreamContent(content);
+        private static void SetRequestHeaders(HttpWebRequest request, IHeaderDictionary requestHeaders) {
+            // copy headers to avoid messing with original request headers
+            Dictionary<string, string> headers = new Dictionary<string, string>();
+            foreach (var pair in requestHeaders) {
+                headers.Add(pair.Key, pair.Value);
```

Comment:
It looks like `pair.Value` here is an instance of `StringValues`, rather than `string`, and so can contain multiple values, but has an implicit conversion operator to `string`. I'm not entirely clear on what this does if there are actually multiple values, but I suspect it might either throw, or drop all values except for the first, neither of which is good.